api/src/main.ts

```diff
@@ -8,13 +9,18 @@ import { productListHandler } from "./routes/product-list"
 import { productDetailHandler } from "./routes/product-detail"
 import { categoryListHandler } from "./routes/category-list"
 
-import { PORT } from "./constants/env"
+import { METABASE_INSTANCE_URL, PORT, VERCEL_ENV } from "./constants/env"
 import { pg } from "./utils/db"
 
 const app = express()
-setupMiddleware(app)
+const isHosted = !!VERCEL_ENV
+if (isHosted) {
+  app.set("trust proxy", 1)
+}
 
 const router = Router()
+setupMiddleware(router)
```

Comment:
When the middle is used for `createProxyMiddleware` it failed. I didn't bother to debug it, but it seems we don't expect to setup any middleware for that proxy one.